### PR TITLE
[Spring Cleanup] Remove Spring dependencies in the email template management API

### PR DESCRIPTION
**Pull request by NipuniBhagya**
## Related Issue(s)

- https://github.com/wso2/product-is/issues/21159

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.email.template/org.wso2.carbon.identity.rest.api.server.email.template.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/email/template/v1/factories/ServerEmailTemplatesServiceFactory.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+/**
+ * Factory class for ServerEmailTemplatesService.
+ */
+public class ServerEmailTemplatesServiceFactory {
+
+    private static final ServerEmailTemplatesService SERVICE;
+
+    static {
+        EmailTemplateManager emailTemplateManager = EmailTemplatesServiceHolder.getEmailTemplateManager();
+
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.api.server.email.template/org.wso2.carbon.identity.rest.api.server.email.template.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/email/template/v1/factories/ServerEmailTemplatesServiceFactory.java:34]: extra line

File: components/org.wso2.carbon.identity.api.server.email.template/org.wso2.carbon.identity.rest.api.server.email.template.v2/pom.xml [hunk 2]
```diff
@@ -174,5 +170,11 @@
             <groupId>org.wso2.carbon.identity.server.api</groupId>
             <artifactId>org.wso2.carbon.identity.api.server.email.template.common</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.identity.server.api</groupId>
+            <artifactId>org.wso2.carbon.identity.rest.api.server.email.template.v1</artifactId>
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.api.server.email.template/org.wso2.carbon.identity.rest.api.server.email.template.v2/pom.xml:175]: why we need this?

File: components/org.wso2.carbon.identity.api.server.email.template/org.wso2.carbon.identity.rest.api.server.email.template.v2/pom.xml [hunk 3]
```diff
@@ -174,5 +170,11 @@
...
+            <version>1.2.228-SNAPSHOT</version>
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.api.server.email.template/org.wso2.carbon.identity.rest.api.server.email.template.v2/pom.xml:176]: remove version from component level pom file

**Comment by lashinijay** [comment]
Comments will be addressed with https://github.com/wso2/identity-api-server/pull/756

**Review by nilasini — APPROVED** [review]

**Pull request merged**